src/workerd/api/unsafe.c++

```diff
@@ -99,4 +90,13 @@ UnsafeEval::UnsafeEvalFunction UnsafeEval::newAsyncFunction(
   return KJ_ASSERT_NONNULL(handler.tryUnwrap(js, result.As<v8::Function>()));
 }
 
-}  // namespace workerd::api
+jsg::JsValue UnsafeEval::newWasmModule(jsg::Lock& js, kj::Array<kj::byte> src) {
+  js.setAllowEval(true);
+  KJ_DEFER(js.setAllowEval(false));
+
+  auto maybeWasmModule = v8::WasmModuleObject::Compile(
```

Comment:
should we reuse compileWasmGlobal here? It also has something about background threads in it.

Reply:
The `compileWasmGlobal` method is currently defined in `workerd-api.c++` for workerd. The internal repo has its own copy of the function. We could likely move that into jsg and use the one impl in every case. I wouldn't block this PR on it tho.

Reply:
how about `AllowV8BackgroundThreadsScope`? Do we need it here?

Reply:
We probably do, yes.